packages/glimmer-reference/lib/iterable.ts

```diff
@@ -86,18 +91,18 @@ export class IterationArtifacts {
     return node && node.seen;
   }
 
-  append(item: OpaqueIterationItem): ListItem {
+  append(item: OpaqueIterationItem, position: number | FIXME<'user string to InternedString'>): ListItem {
     let { map, list, iterable } = this;
+    let node = map[item.key] = new ListItem(iterable, item, position);
 
-    let node = map[item.key] = new ListItem(iterable, item);
     list.append(node);
     return node;
   }
 
-  insertBefore(item: OpaqueIterationItem, reference: ListItem): ListItem {
+  insertBefore(item: OpaqueIterationItem, reference: ListItem, position: number | FIXME<'user string to InternedString'>): ListItem {
     let { map, list, iterable } = this;
 
-    let node = map[item.key] = new ListItem(iterable, item);
+    let node = map[item.key] = new ListItem(iterable, item, position);
```

Comment:
why are we accepting a ListItem and a position, and then creating a new ListItem?